supermarq-benchmarks/supermarq/benchmarks/bit_code.py

```diff
@@ -20,7 +20,12 @@ class BitCode(Benchmark):
 
     def __init__(self, num_data_qubits: int, num_rounds: int, bit_state: List[int]) -> None:
         if len(bit_state) != num_data_qubits:
-            raise ValueError("The length of `bit_state` must match the number of data qubits")
+            raise ValueError("The length of `bit_state` must match the number of data qubits.")
+        if not isinstance(bit_state, List):
+            raise ValueError("`bit_state` must be a List[int].")
+        else:
+            if not isinstance(bit_state[0], int):
```

Comment:
Is this supposed to only validate the first entry of `bit_state` as an `int`?

Reply:
Haha I think there's probably a more elegant way of doing this check. Every element of the list should be an `int` but it seemed clunky to me to iterate through every single element.

Reply:
Using `set` is efficient enough, so I think this could solve that issue: 
```suggestion
            if not set(phase_state).issubset({0, 1})::
```